File: backend/src/main/java/de/interaapps/pastefy/model/database/PasteTag.java
package de.interaapps.pastefy.model.database;

import org.javawebstack.orm.Model;
import org.javawebstack.orm.annotation.Column;
import org.javawebstack.orm.annotation.Filterable;
import org.javawebstack.orm.annotation.Table;

@Table("paste_tags")
public class PasteTag extends Model {
    @Column
    private int id;


    @Column(size = 8)
    @Filterable
    public String paste;

    @Column(size = 30)
    @Filterable
    public String tag;

    @Override
    public void save() {
        if (tag.length() > 30)
            tag = tag.substring(0, 30);

        super.save();
    }
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/entities/PasteTag.kt
package de.interaapps.pastefy.entities

import jakarta.persistence.*

@Entity
@Table(
    name = "pastefy_paste_tags",
    indexes = [
        Index(name = "pastefy_paste_tags_paste_tag_index", columnList = "paste, tag"),
    ],
)
class PasteTag(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    @Column(length = 8, nullable = false) var paste: String = "",
    @Column(length = 30, nullable = false) var tag: String = "",
) {
    @PrePersist
    @PreUpdate
    fun truncateTag() {
        if (tag.length > 30) tag = tag.substring(0, 30)
    }
}
